socket-io.js

```diff
@@ -378,6 +425,43 @@ function SocketIO(server, system) {
             }
         });
     }
+    /**
+     * Symbol used to indicate whetter the socket can be written
+     */
+    const WritingState = Symbol('writing');
+    /**
+     * Symbol used hold a queue of pending writes
+     */
+    const PendingWrites = Symbol('pending');
+    /**
+     * Write to socket handling backpressure
+     */
+    function writeToSocket(stream, data) {
+        const pending = stream[PendingWrites] = (stream[PendingWrites] || []);
```

Comment:
Hi, funny,  i have never come across using the `Symbol` type before.   Is the point  of this to avoid creating custom named variables  on the local `stream` object ?  So to avoid using a variable name as a unique key ?

Reply:
It allows to create non enumerable properties on an object (meaning it will not be returned by Object.entries and other ways to list the object keys) and unlike a custom property name it can't collision with other part of the code, because event if you create other symbol with same name is not considered to be the same symbol.

I can't remember when I learnt about it but I always found it very useful.